### ISSUE TITLE
lock file already exists causing hang

### ISSUE THREAD

**Issue by stscieisenhamer** [spacetelescope/crds#311]
While running the calwebb_spec2 pipeline, I interrupted the process. Then, went to run it again, and it hung after instantiating all the steps.  When interrupting this process, the following traceback occurs:
```
2017-08-04 11:37:26,436 - stpipe.Spec2Pipeline - INFO - Spec2Pipeline instance created.
2017-08-04 11:37:26,436 - stpipe.Spec2Pipeline - INFO - Spec2Pipeline instance created.
2017-08-04 11:37:26,436 - stpipe.Spec2Pipeline - INFO - Spec2Pipeline instance created.
2017-08-04 11:37:26,436 - stpipe.Spec2Pipeline - INFO - Spec2Pipeline instance created.
2017-08-04 11:37:26,441 - stpipe.Spec2Pipeline.extract_1d - INFO - Extract1dStep instance created.
2017-08-04 11:37:26,441 - stpipe.Spec2Pipeline.extract_1d - INFO - Extract1dStep instance created.
2017-08-04 11:37:26,441 - stpipe.Spec2Pipeline.extract_1d - INFO - Extract1dStep instance created.
2017-08-04 11:37:26,441 - stpipe.Spec2Pipeline.extract_1d - INFO - Extract1dStep instance created.
2017-08-04 11:37:26,444 - stpipe.Spec2Pipeline.straylight - INFO - StraylightStep instance created.
2017-08-04 11:37:26,444 - stpipe.Spec2Pipeline.straylight - INFO - StraylightStep instance created.
2017-08-04 11:37:26,444 - stpipe.Spec2Pipeline.straylight - INFO - StraylightStep instance created.
2017-08-04 11:37:26,444 - stpipe.Spec2Pipeline.straylight - INFO - StraylightStep instance created.
2017-08-04 11:37:26,447 - stpipe.Spec2Pipeline.assign_wcs - INFO - AssignWcsStep instance created.
2017-08-04 11:37:26,447 - stpipe.Spec2Pipeline.assign_wcs - INFO - AssignWcsStep instance created.
2017-08-04 11:37:26,447 - stpipe.Spec2Pipeline.assign_wcs - INFO - AssignWcsStep instance created.
2017-08-04 11:37:26,447 - stpipe.Spec2Pipeline.assign_wcs - INFO - AssignWcsStep instance created.
2017-08-04 11:37:26,449 - stpipe.Spec2Pipeline.srctype - INFO - SourceTypeStep instance created.
2017-08-04 11:37:26,449 - stpipe.Spec2Pipeline.srctype - INFO - SourceTypeStep instance created.
2017-08-04 11:37:26,450 - stpipe.Spec2Pipeline.srctype - INFO - SourceTypeStep instance created.
2017-08-04 11:37:26,450 - stpipe.Spec2Pipeline.srctype - INFO - SourceTypeStep instance created.
2017-08-04 11:37:26,452 - stpipe.Spec2Pipeline.fringe - INFO - FringeStep instance created.
2017-08-04 11:37:26,452 - stpipe.Spec2Pipeline.fringe - INFO - FringeStep instance created.
2017-08-04 11:37:26,453 - stpipe.Spec2Pipeline.fringe - INFO - FringeStep instance created.
2017-08-04 11:37:26,452 - stpipe.Spec2Pipeline.fringe - INFO - FringeStep instance created.
2017-08-04 11:37:26,454 - stpipe.Spec2Pipeline.bkg_subtract - INFO - BackgroundStep instance created.
2017-08-04 11:37:26,455 - stpipe.Spec2Pipeline.bkg_subtract - INFO - BackgroundStep instance created.
2017-08-04 11:37:26,455 - stpipe.Spec2Pipeline.bkg_subtract - INFO - BackgroundStep instance created.
2017-08-04 11:37:26,456 - stpipe.Spec2Pipeline.bkg_subtract - INFO - BackgroundStep instance created.
2017-08-04 11:37:26,457 - stpipe.Spec2Pipeline.extract_2d - INFO - Extract2dStep instance created.
2017-08-04 11:37:26,457 - stpipe.Spec2Pipeline.extract_2d - INFO - Extract2dStep instance created.
2017-08-04 11:37:26,458 - stpipe.Spec2Pipeline.extract_2d - INFO - Extract2dStep instance created.
2017-08-04 11:37:26,458 - stpipe.Spec2Pipeline.extract_2d - INFO - Extract2dStep instance created.
2017-08-04 11:37:26,460 - stpipe.Spec2Pipeline.cube_build - INFO - CubeBuildStep instance created.
2017-08-04 11:37:26,461 - stpipe.Spec2Pipeline.cube_build - INFO - CubeBuildStep instance created.
2017-08-04 11:37:26,462 - stpipe.Spec2Pipeline.cube_build - INFO - CubeBuildStep instance created.
2017-08-04 11:37:26,462 - stpipe.Spec2Pipeline.cube_build - INFO - CubeBuildStep instance created.
2017-08-04 11:37:26,463 - stpipe.Spec2Pipeline.pathloss - INFO - PathLossStep instance created.
2017-08-04 11:37:26,463 - stpipe.Spec2Pipeline.pathloss - INFO - PathLossStep instance created.
2017-08-04 11:37:26,464 - stpipe.Spec2Pipeline.pathloss - INFO - PathLossStep instance created.
2017-08-04 11:37:26,465 - stpipe.Spec2Pipeline.pathloss - INFO - PathLossStep instance created.
2017-08-04 11:37:26,465 - stpipe.Spec2Pipeline.imprint_subtract - INFO - ImprintStep instance created.
2017-08-04 11:37:26,466 - stpipe.Spec2Pipeline.imprint_subtract - INFO - ImprintStep instance created.
2017-08-04 11:37:26,467 - stpipe.Spec2Pipeline.imprint_subtract - INFO - ImprintStep instance created.
2017-08-04 11:37:26,467 - stpipe.Spec2Pipeline.imprint_subtract - INFO - ImprintStep instance created.
2017-08-04 11:37:26,468 - stpipe.Spec2Pipeline.photom - INFO - PhotomStep instance created.
2017-08-04 11:37:26,468 - stpipe.Spec2Pipeline.photom - INFO - PhotomStep instance created.
2017-08-04 11:37:26,469 - stpipe.Spec2Pipeline.photom - INFO - PhotomStep instance created.
2017-08-04 11:37:26,470 - stpipe.Spec2Pipeline.photom - INFO - PhotomStep instance created.
2017-08-04 11:37:26,471 - stpipe.Spec2Pipeline.resample_spec - INFO - ResampleSpecStep instance created.
2017-08-04 11:37:26,472 - stpipe.Spec2Pipeline.resample_spec - INFO - ResampleSpecStep instance created.
2017-08-04 11:37:26,473 - stpipe.Spec2Pipeline.resample_spec - INFO - ResampleSpecStep instance created.
2017-08-04 11:37:26,473 - stpipe.Spec2Pipeline.resample_spec - INFO - ResampleSpecStep instance created.
2017-08-04 11:37:26,474 - stpipe.Spec2Pipeline.flat_field - INFO - FlatFieldStep instance created.
2017-08-04 11:37:26,475 - stpipe.Spec2Pipeline.flat_field - INFO - FlatFieldStep instance created.
2017-08-04 11:37:26,476 - stpipe.Spec2Pipeline.flat_field - INFO - FlatFieldStep instance created.
2017-08-04 11:37:26,476 - stpipe.Spec2Pipeline.flat_field - INFO - FlatFieldStep instance created.
2017-08-04 11:37:26,598 - stpipe.Spec2Pipeline - INFO - Step Spec2Pipeline running with args ('level2a/jw93135011001_02102_00001_nrs2_rate.fits',).
2017-08-04 11:37:26,598 - stpipe.Spec2Pipeline - INFO - Step Spec2Pipeline running with args ('level2a/jw93135011001_02104_00001_nrs2_rate.fits',).
2017-08-04 11:37:26,598 - stpipe.Spec2Pipeline - INFO - Step Spec2Pipeline running with args ('level2a/jw93135011001_02101_00001_nrs2_rate.fits',).
2017-08-04 11:37:26,598 - stpipe.Spec2Pipeline - INFO - Step Spec2Pipeline running with args ('level2a/jw93135011001_0210b_00001_nrs1_rate.fits',).
2017-08-04 11:37:26,600 - stpipe.Spec2Pipeline - INFO - Starting calwebb_spec2 ...
2017-08-04 11:37:26,600 - stpipe.Spec2Pipeline - INFO - Starting calwebb_spec2 ...
2017-08-04 11:37:26,600 - stpipe.Spec2Pipeline - INFO - Starting calwebb_spec2 ...
2017-08-04 11:37:26,600 - stpipe.Spec2Pipeline - INFO - Starting calwebb_spec2 ...
2017-08-04 11:37:27,210 - stpipe.Spec2Pipeline - INFO - Processing product level2a/jw93135011001_02102_00001_nrs2
2017-08-04 11:37:27,212 - stpipe.Spec2Pipeline - INFO - Working on input level2a/jw93135011001_02102_00001_nrs2_rate.fits ...
2017-08-04 11:37:27,222 - stpipe.Spec2Pipeline - INFO - Processing product level2a/jw93135011001_0210b_00001_nrs1
2017-08-04 11:37:27,224 - stpipe.Spec2Pipeline - INFO - Working on input level2a/jw93135011001_0210b_00001_nrs1_rate.fits ...
2017-08-04 11:37:27,227 - stpipe.Spec2Pipeline - INFO - Processing product level2a/jw93135011001_02101_00001_nrs2
2017-08-04 11:37:27,229 - stpipe.Spec2Pipeline - INFO - Working on input level2a/jw93135011001_02101_00001_nrs2_rate.fits ...
2017-08-04 11:37:27,230 - stpipe.Spec2Pipeline - INFO - Processing product level2a/jw93135011001_02104_00001_nrs2
2017-08-04 11:37:27,233 - stpipe.Spec2Pipeline - INFO - Working on input level2a/jw93135011001_02104_00001_nrs2_rate.fits ...
Process ForkPoolWorker-2:
Process ForkPoolWorker-1:
Process ForkPoolWorker-4:
Process ForkPoolWorker-3:
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/linklockfile.py", line 31, in acquire
    os.link(self.unique_name, self.lock_file)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/linklockfile.py", line 31, in acquire
    os.link(self.unique_name, self.lock_file)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/linklockfile.py", line 31, in acquire
    os.link(self.unique_name, self.lock_file)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/linklockfile.py", line 31, in acquire
    os.link(self.unique_name, self.lock_file)
FileExistsError: [Errno 17] File exists: '/tmp/inanna2.stsci.edu-cf39c3c0.36594319212574973908563' -> '/tmp/crds.cache.lock.lock'
FileExistsError: [Errno 17] File exists: '/tmp/inanna2.stsci.edu-cf39c3c0.36574319212574973908563' -> '/tmp/crds.cache.lock.lock'
FileExistsError: [Errno 17] File exists: '/tmp/inanna2.stsci.edu-cf39c3c0.36564319212574973908563' -> '/tmp/crds.cache.lock.lock'
FileExistsError: [Errno 17] File exists: '/tmp/inanna2.stsci.edu-cf39c3c0.36584319212574973908563' -> '/tmp/crds.cache.lock.lock'

During handling of the above exception, another exception occurred:


During handling of the above exception, another exception occurred:


During handling of the above exception, another exception occurred:


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "<ipython-input-5-db23127d4248>", line 5, in spec2run
    output_dir='level2b'
  File "<ipython-input-5-db23127d4248>", line 5, in spec2run
    output_dir='level2b'
  File "<ipython-input-5-db23127d4248>", line 5, in spec2run
    output_dir='level2b'
  File "<ipython-input-5-db23127d4248>", line 5, in spec2run
    output_dir='level2b'
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 479, in call
    return instance.run(*args)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 479, in call
    return instance.run(*args)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 479, in call
    return instance.run(*args)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 479, in call
    return instance.run(*args)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 361, in run
    result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 361, in run
    result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 361, in run
    result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 361, in run
    result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 83, in process
    asn.filename
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 83, in process
    asn.filename
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 83, in process
    asn.filename
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 83, in process
    asn.filename
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 137, in process_exposure_product
    input = self.assign_wcs(input)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 137, in process_exposure_product
    input = self.assign_wcs(input)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 137, in process_exposure_product
    input = self.assign_wcs(input)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 137, in process_exposure_product
    input = self.assign_wcs(input)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 336, in run
    self._precache_reference_files(args[0])
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 336, in run
    self._precache_reference_files(args[0])
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 336, in run
    self._precache_reference_files(args[0])
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 336, in run
    self._precache_reference_files(args[0])
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 549, in _precache_reference_files
    self._precache_reference_files_opened(model)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 549, in _precache_reference_files
    self._precache_reference_files_opened(model)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 549, in _precache_reference_files
    self._precache_reference_files_opened(model)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 549, in _precache_reference_files
    self._precache_reference_files_opened(model)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 573, in _precache_reference_files_opened
    self._precache_reference_files_impl(model_or_container)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 573, in _precache_reference_files_opened
    self._precache_reference_files_impl(model_or_container)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 573, in _precache_reference_files_opened
    self._precache_reference_files_impl(model_or_container)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 573, in _precache_reference_files_opened
    self._precache_reference_files_impl(model_or_container)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 591, in _precache_reference_files_impl
    crds_refs = crds_client.get_multiple_reference_paths(model, fetch_types)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 591, in _precache_reference_files_impl
    crds_refs = crds_client.get_multiple_reference_paths(model, fetch_types)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 591, in _precache_reference_files_impl
    crds_refs = crds_client.get_multiple_reference_paths(model, fetch_types)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 591, in _precache_reference_files_impl
    crds_refs = crds_client.get_multiple_reference_paths(model, fetch_types)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/crds_client.py", line 97, in get_multiple_reference_paths
    with crds_cache_locking.get_cache_lock():
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/crds_client.py", line 97, in get_multiple_reference_paths
    with crds_cache_locking.get_cache_lock():
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/crds_client.py", line 97, in get_multiple_reference_paths
    with crds_cache_locking.get_cache_lock():
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/jwst-0.7.8rc1.dev296-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/crds_client.py", line 97, in get_multiple_reference_paths
    with crds_cache_locking.get_cache_lock():
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/__init__.py", line 197, in __enter__
    self.acquire()
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/__init__.py", line 197, in __enter__
    self.acquire()
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/__init__.py", line 197, in __enter__
    self.acquire()
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/__init__.py", line 197, in __enter__
    self.acquire()
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/crds/core/crds_cache_locking.py", line 72, in acquire
    result = super(CrdsLockFile, self).acquire(*args, **keys)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/crds/core/crds_cache_locking.py", line 72, in acquire
    result = super(CrdsLockFile, self).acquire(*args, **keys)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/crds/core/crds_cache_locking.py", line 72, in acquire
    result = super(CrdsLockFile, self).acquire(*args, **keys)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/crds/core/crds_cache_locking.py", line 72, in acquire
    result = super(CrdsLockFile, self).acquire(*args, **keys)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/linklockfile.py", line 50, in acquire
    time.sleep(timeout is not None and timeout / 10 or 0.1)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/linklockfile.py", line 50, in acquire
    time.sleep(timeout is not None and timeout / 10 or 0.1)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/linklockfile.py", line 50, in acquire
    time.sleep(timeout is not None and timeout / 10 or 0.1)
  File "/Users/eisenham/anaconda3/envs/pipeline/lib/python3.5/site-packages/lockfile/linklockfile.py", line 50, in acquire
    time.sleep(timeout is not None and timeout / 10 or 0.1)
KeyboardInterrupt
KeyboardInterrupt
KeyboardInterrupt
KeyboardInterrupt
---------------------------------------------------------------------------
KeyboardInterrupt                         Traceback (most recent call last)
<ipython-input-13-7a3c5c492271> in <module>()
      4         (
      5             product['members'][0]['expname']
----> 6             for product in lvl2_asn['products']
      7         )
      8     )

~/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py in map(self, func, iterable, chunksize)
    258         in a list that is returned.
    259         '''
--> 260         return self._map_async(func, iterable, mapstar, chunksize).get()
    261 
    262     def starmap(self, func, iterable, chunksize=None):

~/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py in get(self, timeout)
    600 
    601     def get(self, timeout=None):
--> 602         self.wait(timeout)
    603         if not self.ready():
    604             raise TimeoutError

~/anaconda3/envs/pipeline/lib/python3.5/multiprocessing/pool.py in wait(self, timeout)
    597 
    598     def wait(self, timeout=None):
--> 599         self._event.wait(timeout)
    600 
    601     def get(self, timeout=None):

~/anaconda3/envs/pipeline/lib/python3.5/threading.py in wait(self, timeout)
    547             signaled = self._flag
    548             if not signaled:
--> 549                 signaled = self._cond.wait(timeout)
    550             return signaled
    551 

~/anaconda3/envs/pipeline/lib/python3.5/threading.py in wait(self, timeout)
    291         try:    # restore state no matter what (e.g., KeyboardInterrupt)
    292             if timeout is None:
--> 293                 waiter.acquire()
    294                 gotit = True
    295             else:

KeyboardInterrupt: 
```

**Comment by stscieisenhamer** [comment]
@jaytmiller i've filed another one

**Comment by jaytmiller** [comment]
I think this problem may be not catching / setting up Keyboard interrupt appropriately before multiprocessing pool.  Bear with me:

https://stackoverflow.com/questions/11312525/catch-ctrlc-sigint-and-exit-multiprocesses-gracefully-in-python

The lockfile creation did fail,  but the Keyboard interrupt is happening inside the multiprocessing pool wait.   I saw this during HSI I think and the key was to do a join as is seen farther down in the web link.   (So I think maybe *any* exception occuring inside your code may get stuck.  Read the thread and see what you think.)

There is a related mystery though of how the temp lock file was already created.  That's bogus.

**Comment by stscieisenhamer** [comment]
The keyboard interrupt was done on purpose to get out of the hang situation; the interrupt was not causing the hang.

However, to note which I also noted in the other call, I was running this on linux machines all last week without issue. I'll try the run on my mac today, with the latest code, and see if I can reproduce.

**Comment by stscieisenhamer** [comment]
OK. so last week I was not doing any multiprocessing. I've just confirmed, on my mac, that the issue, with today's conda-dev environment, still exists.  To clarify, with CRDS_PATH set to a local folder, no other CRDS environmentals are defined, and the local folder is empty, the following script will cause the hang condition. Run this script in the test data folder I had sent instructions to download earlier; ping me again if you need that info. If I pre-load the local cache using bestrefs, all works fine.

```
from multiprocessing import Pool
from jwst.pipeline import Spec2Pipeline
from jwst.associations import load_asn
from jwst.associations.lib.update_path import update_path

with open('jw93135-o011_20170721t013702_spec2_002_asn.json') as fp:
    lvl2_asn = load_asn(fp)
update_path(lvl2_asn, 'level2a')

with Pool(processes=4, maxtasksperchild=1) as process_pool:
    process_pool.map(
        spec2run,
        [
            product['members'][0]['expname']
            for product in lvl2_asn['products']
        ]
    )
```

**Comment by stscieisenhamer** [comment]
Another fly-in-the-ointment. I have also discovered that no matter what, running the equivalent to the above script in a jupyter notebook causes the hang to occur.

**Comment by stscieisenhamer** [comment]
Just tried this all out on a linux box; all worked fine. Seems just to be os x.

**Comment by stscieisenhamer** [comment]
Should be resolved by commit #93a8f716
